src/python/WMCore/ACDC/DataCollectionService.py

```diff
@@ -93,10 +93,13 @@ def failedJobs(self, failedJobs, useMask=True):
             coll.addFileset(fileset)
             inputFiles = job['input_files']
             for fInfo in inputFiles:
-                if fInfo["merged"] and ("parents" in fInfo) and \
-                   len(fInfo["parents"]) and ("/store/unmerged/" in next(iter(fInfo["parents"]))):
-                    # remove parents files from acdc doucment if they are unmerged files
-                    fInfo["parents"] = []
+                if ("parents" in fInfo) and len(fInfo["parents"]) and ("/store/unmerged/" in next(iter(fInfo["parents"]))):
```

Comment:
updated the code, However, DAO can't be changed since this is used in JobAccountant as well. I am not sure how heavy the load is it just get parent file lfn.  Anyway, we need to first check what is the performance penalty on this.